lib/AST/LookupVisibleDecls.cpp

```diff
@@ -1067,6 +1066,65 @@ void swift::lookupVisibleDecls(VisibleDeclConsumer &Consumer,
   }
 }
 
+void swift::lookupVisibleDecls(VisibleDeclConsumer &Consumer,
+                               const DeclContext *DC,
+                               LazyResolver *TypeResolver,
+                               bool IncludeTopLevel,
+                               SourceLoc Loc) {
+  if (Loc.isInvalid()) {
+    lookupVisibleDeclsImpl(Consumer, DC, TypeResolver, IncludeTopLevel, Loc);
+    return;
+  }
+
+  // Filtering out unusable values.
+  class LocalConsumer : public VisibleDeclConsumer {
+    const SourceManager &SM;
+    SourceLoc Loc;
+    VisibleDeclConsumer &Consumer;
+
+    bool checkLocalValue(ValueDecl *VD, DeclVisibilityKind Reason) {
```

Comment:
Please rename this to something that makes it clear what `true` and `false` return values mean - e.g. `isUsableLocalValue`.